app/models/fraud/indicator.rb

```diff
@@ -96,10 +97,15 @@ def equals(references)
 
     private
 
+    def query_model
+      query_model_name.constantize
+    end
+
     def scoped_records(references)
-      self_reference = query_model_name.underscore == reference
+      compare_model_name = query_model_name.include?("Intake::") ? "Intake" : query_model_name # match from child intake type
```

Comment:
We often have checks that are scoped down to Intake::CtcIntake, which is still a self-reference to intake

Reply:
the object passed into the references block in execute is always going to be "intake", but the reference table is sometimes/often going to be Intake::CtcIntake because we only want to look at Ctc intakes. This would blow up if we tried to compare intake_ctc_intake to intake by attempting to create a query that looks like Intake::CtcIntake.where(intake: intake), but it needs to be Intake::CtcIntake.where(id: intake.id)